tests/unit/test_charm.py

```diff
@@ -1,67 +1,265 @@
+#!/usr/bin/env python3
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more about testing at: https://juju.is/docs/sdk/testing
 
+import json
 import unittest
-from unittest.mock import Mock
+from unittest.mock import Mock, PropertyMock, call, patch
 
-from ops.model import ActiveStatus
-from ops.testing import Harness
+import yaml
+from ops import testing
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
 
-from charm import OperatorTemplateCharm
+from charm import AlertmanagerConfigurerOperatorCharm
 
+testing.SIMULATE_CAN_CONNECT = True
 
-class TestCharm(unittest.TestCase):
+TEST_MULTITENANT_LABEL = "some_test_label"
+TEST_CONFIG = f"""options:
+  multitenant_label:
+    type: string
+    description: |
+      Alertmanager Configurer has been designed to support multiple tenants. In a multitenant
+      Alertmanager Configurer setup, each alert is first routed on the tenancy label, and then
+      the routing tree is distinct for each tenant.
+    default: {TEST_MULTITENANT_LABEL}
+"""
+TEST_ALERTMANAGER_CONFIG_FILE = "/test/rules/dir/config_file.yml"
+TEST_ALERTMANAGER_DEFAULT_CONFIG = """route:
+  receiver: test_receiver
+  group_by:
+  - alertname
+  group_wait: 1234s
+  group_interval: 4321s
+  repeat_interval: 1111h
+receivers:
+- name: test_receiver
+    """
+
+
+class TestAlertmanagerConfigurerOperatorCharm(unittest.TestCase):
+    @patch("charm.KubernetesServicePatch", lambda charm, ports: None)
     def setUp(self):
-        self.harness = Harness(OperatorTemplateCharm)
+        self.harness = testing.Harness(
+            AlertmanagerConfigurerOperatorCharm, config=TEST_CONFIG
+        )
         self.addCleanup(self.harness.cleanup)
+        self.harness.set_leader(True)
```

Comment:
We should probably not set_leader all the time. We probably want to validate some behaviours for non leader units.

Reply:
Yes, there's still you comment about HA (above) pending. I'll be testing this asap and then also the relevant tests will come.

Reply:
Tests for non-leader unit added.